### PR TITLE
docs: update docker login reference

### PR DESCRIPTION
**Pull request by dvdksn**
Signed-off-by: David Karlsson <35727626+dvdksn@users.noreply.github.com>

### CODE REVIEW

File: docs/reference/commandline/login.md [hunk 1]
```diff
@@ -162,19 +157,93 @@ the credentials from the default store.
...
+If you want to authenticate to a self-hosted registry you can specify this by
+adding the server name.
+
+```console
+$ docker login registry.example.com
+```
+
+By default, the `docker login` command assumes that the registry listens on
+port `5000`. If the registry listens on a different port, you can specify it
+by adding the port number to the server name.
```
Comment (thaJeztah) [inline docs/reference/commandline/login.md:208]: I don't think it assumes port 5000, but will default to `443` (TLS) / `80` (non-TLS). (The examples for the registry tend to use port 5000 though, so maybe that's what you assumed here?)

Comment (dvdksn) [reply docs/reference/commandline/login.md:208]: ops yea my bad!

File: docs/reference/commandline/login.md [hunk 2]
```diff
@@ -1,60 +1,51 @@
...
-You can log in to any public or private repository for which you have
-credentials.  When you log in, the command stores credentials in
-`$HOME/.docker/config.json` on Linux or `%USERPROFILE%/.docker/config.json` on
-Windows, via the procedure described below.
+You can authenticate to a registry using a username and password or access
+token. Docker Hub also supports a web-based login flow, which signs you in to
+your Docker account without entering your password. For Docker Hub, the `docker
+login` command uses a device code flow by default, unless the `--username` flag
+is specified. The device code flow is a secure way to sign in. See
+[Authenticate to Docker Hub using device code](#authenticate-to-docker-hub-using-device-code).
```
[the remark this answers is not in the document]
Comment (dvdksn) [reply docs/reference/commandline/login.md]: Yeah we could... @laurazard do we have some special sauce for handling login differently for non-TTY cases?

[the remark this answers is not in the document]
Comment (laurazard) [reply docs/reference/commandline/login.md]: Ah, we don't. With `--username`, without setting the password we do error out if non-tty:
```console
$ docker login -u bork 0>/dev/null
Error: Cannot perform an interactive login from a non TTY device
```
But with the device-code flow, we don't:
```console
$ docker login 0>/dev/null

USING WEB BASED LOGIN
To sign in with credentials on the command line, use 'docker login -u <username>'

Your one-time device confirmation code is: JZJG-ZWNM
Press ENTER to open your browser or submit your device code here: https://login.docker.com/activate

Waiting for authentication in the browser…
```

Although in the first case, there's no way for the user to login when non-interactive, but with the device-code flow a user could open the URL and login on another device. Maybe we should though 🤔

[the remark this answers is not in the document]
Comment (laurazard) [reply docs/reference/commandline/login.md]: Ah, we don't. With `--username`, without setting the password we do error out if non-tty:
```console
$ docker login -u bork 0>/dev/null
Error: Cannot perform an interactive login from a non TTY device
```
But with the device-code flow, we don't:
```console
$ docker login 0>/dev/null

USING WEB BASED LOGIN
To sign in with credentials on the command line, use 'docker login -u <username>'

Your one-time device confirmation code is: <--->
Press ENTER to open your browser or submit your device code here: https://login.docker.com/activate

Waiting for authentication in the browser…
```

Although in the first case, there's no way for the user to login when non-interactive, but with the device-code flow a user could open the URL and login on another device. Maybe we should though 🤔

File: docs/reference/commandline/login.md [hunk 3]
```diff
@@ -1,60 +1,51 @@
...
-or log-files.
-
-The following example reads a password from a file, and passes it to the
-`docker login` command using `STDIN`:
-
-```console
-$ cat ~/my_password.txt | docker login --username foo --password-stdin
-```
+You can authenticate to any public or private registry for which you have
+credentials. Authentication is required for pulling and pushing images. Other
```
Comment (laurazard) [inline docs/reference/commandline/login.md:23]: Authentication isn't always required (pulling public images from certain registries).
```suggestion
credentials. Authentication may be required for pulling and pushing images. Other
```

File: docs/reference/commandline/login.md [hunk 4]
```diff
@@ -162,19 +157,93 @@ the credentials from the default store.
...
-    "unicorn.example.io": "vcbait"
+    "myregistry.example.com": "secretservice",
+    "docker.internal.example": "pass",
   }
 }
 ```
 
+## Examples
+
+### Authenticate to Docker Hub using device code
```
Comment (laurazard) [inline docs/reference/commandline/login.md:173]:
```suggestion
### Authenticate to Docker Hub with web-based login
```

**Review by laurazard — APPROVED** [review]
LGTM! Thanks for this @dvdksn

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@dvdksn can you open a cherry-pick as well?

**Pull request merged**